src/ORM/Association/BelongsToMany.php

```diff
@@ -387,7 +387,9 @@ protected function _appendNotMatching($query, $options)
             ->where($options['conditions'])
             ->andWhere($this->junctionConditions());
 
-        $subquery = $options['queryBuilder']($subquery);
+        if (!empty($options['queryBuilder'])) {
```

Comment:
I guess it can't be in the context of `notMatching()`, but since `_appendNotMatching()` doesn't enforce defaults for it, it generally could be undefined.